src/pybind/mgr/dashboard/frontend/src/app/shared/directives/update-scope.directive.ts

```diff
@@ -0,0 +1,34 @@
+import { AfterViewInit, Directive, ElementRef, Input, Optional } from '@angular/core';
+import { Permissions } from '../models/permissions';
+import { AuthStorageService } from '../services/auth-storage.service';
+import { PermissionScopeDirective } from './permission-scope.directive';
+
+@Directive({
+  selector: 'input:not([cdNoUpdateScope]), select:not([cdNoUpdateScope]), [cdUpdateScope]'
+})
+export class UpdateScopeDirective implements AfterViewInit {
+  private permissions: Permissions;
+  private service_name: keyof Permissions;
+
+  @Input() disabled: boolean;
+
+  constructor(
+    @Optional() private scope: PermissionScopeDirective,
+    private authStorageService: AuthStorageService,
+    private elementRef: ElementRef
```

Comment:
Didn't @Hostbinding work here finally? In Angular [ElementRef is discouraged for security risks](https://angular.io/api/core/ElementRef) in favor of non-direct DOM manipulation (@Hostbinding or, as a last resort, Rendered2 service).

Reply:
@HostBinding works here for disabling but it somehow fails the make check everytime. Thats why I went around and used ElementRef. 
I'll see what I can do with Rendered2 service.